File: python/teemo_attacking.py
from typing import List


class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        poisonDuration = 0
        index, current = 0, timeSeries[0] - 1
        while index < len(timeSeries):
            poisonDuration += duration - (current - timeSeries[index] + 1)
            current = max(
                timeSeries[index] + duration - 1,
                timeSeries[index + 1] - 1 if index < len(timeSeries) - 1 else 0
            )
            index += 1
        return poisonDuration

File: src/TeemoAttacking.java
public class TeemoAttacking {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int poisonDuration = 0;
        for (int i = 0, current = timeSeries[0] - 1 ; i < timeSeries.length; i++) {
            poisonDuration += duration - Math.max(0, Math.max(current, timeSeries[i] - 1) - timeSeries[i] + 1);
            current = timeSeries[i] + duration - 1;
        }
        return poisonDuration;
    }
}
